### ISSUE TITLE
Social Links: Angular Error

### ISSUE THREAD

**Issue by 37Stars** [leekelleher/umbraco-contentment#422]
### Which Contentment version are you using?

5.0.2

### Which Umbraco version are you using? For example: 13.2.1 - don't just write v13

Umbraco 13.5.2

### Bug summary

I get multiple Angular errors when the Social Network dialog appears and I cannot close the dialog. The last error in the list happens when you click the Close link.

![Screenshot 2024-11-14 172818](https://github.com/user-attachments/assets/63a17145-6ce9-4c8a-91de-0169d166bf0d)

### Steps to reproduce

I didn't do anything out of the ordinary. I simply created a new DataType using Social Links as the property editor. Added that Data Type to an element that. The element was placed on a Document Type. When I attempt to select a Social Network, I get the errors in the screenshot above.

### Do you have Umbraco ModelsBuilder enabled?

- [X] Yes, it is enabled.

### What browsers are you seeing the problem on?

Chrome, Edge, Firefox

**Comment by leekelleher** [comment]
@37Stars Thanks for raising this issue.

I'm not able to reproduce the bug following your steps.

From looking at the JS errors, specifically the `Cannot read properties of null (reading 'network')` error, this suggests that there may be an issue with the "Social networks" configuration in the data-type. I'm wondering if it's possible that one of the items is missing a `network` value? _(Although I've tested this scenario and the UI should still work with an empty/null/undefined value)_ 🤔 

Whilst reviewing the UI code, I have noticed that it could be more robust, so I'll be adding some more value checks in there.
